observations/obs_converters/Ameriflux/fluxnetfull_to_obs.f90

```diff
@@ -0,0 +1,1076 @@
+   call error_handler(E_MSG, source, string1, &
+                      text2=string2,text3=string3)
+
+   if (time_resolution == 'DD' .or. time_resolution == 'MM' .or. &
+       time_resolution == 'WW') then
+      write(string1,*) 'ERROR!: gap filling can only be turned off for native time'
+      write(string2,*) 'resolution data (HH, HR)'
+      write(string3,*) 'Coarser time resolution (DD, WW, MM)  must have gap_filled = .true.'
+      call error_handler(E_ERR, source, string1, &
+                      text2=string2,text3=string3)
+   endif
+
+endif
+
+
+! Modify values to towerdata strings based on user input.nml
+! 1) time_resolution: DD(daily),MM(monthly),YY(yearly) uses 'TIMESTAMP' header
+
+if (time_resolution == 'DD' .or. time_resolution == 'MM' .or. &
+    time_resolution == 'YY') then
+
+  tower%startstring    = 'TIMESTAMP'
+  tower%endstring      = 'TIMESTAMP'
+  res                  = .false.  
+
+  write(string1, *) 'Time resolution is set to =', time_resolution
+  write(string2, *) 'Using TIMESTAMP to set DART observation time'
+  if (verbose) call error_handler(E_MSG,'fluxnetfull_to_obs',string1,text2=string2)
+
+elseif (time_resolution == 'HH' .or. time_resolution == 'HR' .or. &
+        time_resolution == 'WW') then
+
+  res = .true.
+  write(string1, *) 'Time resolution is set to =', time_resolution
+  write(string2, *) 'Using TIMESTAMP_START and TIMESTAMP_END to set: DART observation time'
+  if (verbose) call error_handler(E_MSG,'fluxnetfull_to_obs',string1,text2=string2)
+
+else
+  write(string1,*) 'time_resolution set incorrectly within input.nml'
+  write(string2,*) 'time_resolution must be HR,HH,DD,WW,MM, or YY'
+  call error_handler(E_ERR, source, string1,text2=string2)
+
+endif
+
+! 2) energy_balance: .true. changes sensible and latent heat strings
+!    Note: There are no qc values for energy_balance correction
+!    The qc values are manually set later in code 
+if (energy_balance .eqv. .true.) then
+
+   tower%lestring       = 'LE_CORR'
+   tower%leUNCstring    = 'LE_CORR_JOINTUNC'
+   tower%hstring        = 'H_CORR'
+   tower%hUNCstring     = 'H_CORR_JOINTUNC'
+   write(string1,*) 'WARNING! Energy balance correction data turned on for LE and H'
+   write(string2,*) 'Check to make sure LE_CORR and H_CORR data is not missing'
+   call error_handler(E_MSG, source, string1,text2=string2)
+else
+
+   write(string1,*) 'Standard LE (LE_F_MDS) and H (H_F_MDS) data being used'
+   call error_handler(E_MSG, source, string1)
+endif
+
+! Specify the maximum number of observations in the input file,
+! but only the actual number created will be written out.
+! Each line has 5 flux observations available.
+! Each observation in this series will have a single
+! observation value and a quality control flag.  
+! Initialize two empty observations - one to track location
+! in observation sequence - the other is for the new observation.
+
+iunit = open_file(text_input_file, 'formatted', 'read')
```

Comment:
@braczka this call to open_file is missing a corresponding call to close_file

Reply:
Gotcha -- took care of these in latest commits.